fairlearn/metrics/_extra_metrics.py

```diff
@@ -6,38 +6,194 @@
 These are metrics which are not part of `scikit-learn`.
 """
 
+import numpy as np
 import sklearn.metrics as skm
 
 from ._balanced_root_mean_squared_error import _balanced_root_mean_squared_error  # noqa: F401
 from ._mean_predictions import mean_prediction, _mean_overprediction, _mean_underprediction  # noqa: F401,E501
 from ._selection_rate import selection_rate  # noqa: F401,E501
 
+_TOO_MANY_UNIQUE_Y_VALS = "Must have no more than two unique y values"
+_POS_LABEL_AMBIGUOUS = "Must specify pos_label for degenerate data"
+_NEED_POS_LABEL_IN_Y_VALS = "Must have pos_label in y values"
 
-def true_positive_rate(y_true, y_pred, sample_weight=None):
-    r"""Calculate the true positive rate (also called sensitivity, recall, or hit rate)."""
-    tnr, fpr, fnr, tpr = skm.confusion_matrix(
-        y_true, y_pred, sample_weight=sample_weight, labels=[0, 1], normalize="true").ravel()
+
+def _get_labels_for_confusion_matrix(y_true, y_pred, pos_label):
+    r"""Figure out the labels argument for skm.confusion_matrix.
+
+    This is an internal method used by the true/false positive/negative
+    rate metrics (and hence are restricted to binary data). We compute
+    these using the confusion matrix.
+    This method prepares the `labels` argument of
+    :py:func:`sklearn.metrics.confusion_matrix` based on the
+    user's specifications.
+
+    Parameters
+    ----------
+    y_true : array-like
+        The true values for the metric computation
+
+    y_pred : array-like
+        The predicted values for the metric computation
+
+    pos_label : scalar
+        The value in the true and predicted arrays to treat as positive.
+        This must be set if the two y arrays contain a single unique value.
+        If set to None, then the larger of the two unique values will be used.
+
+    Returns
+    -------
+    list
+        A two element list, consisting of the unique values of the two arrays
+        with the positive label listed last
+    """
+    my_labels = list(np.unique(np.concatenate((y_true, y_pred), axis=None)))
+    if len(my_labels) > 2:
+        raise ValueError(_TOO_MANY_UNIQUE_Y_VALS)
+    if pos_label is None:
+        if len(my_labels) == 1:
+            raise ValueError(_POS_LABEL_AMBIGUOUS)
+    else:
+        if pos_label not in my_labels and len(my_labels) == 2:
+            raise ValueError(_NEED_POS_LABEL_IN_Y_VALS)
+        if len(my_labels) == 2 and my_labels[1] != pos_label:
+            my_labels = list(reversed(my_labels))
+    return my_labels
```

Comment:
this is different from my suggestion in the comment. there are some issues with this default implementation. can you explain why this is preferred?

Reply:
Do you mean:
https://github.com/fairlearn/fairlearn/pull/467#discussion_r439472357
If so, then if the unique values are {1,2} then this does select 2 as the 'positive' case. I've just added some tests to show that.

Or did you mean something else?

Reply:
Let me explain the proposed logic better:
```
if pos_label is None:
   if sorted unique labels are one of [0], [1], [-1], [0, 1], [-1, 1]:
       create `labels` as if pos_label = 1
   else:
       raise ValueError
else:
   ...       
```
The behavior in the current PR differs when unique values are [0], [-1] or ['a', 'b'] and pos_label=None.

Reply:
Ahhh got it. I missed the bit in your initial comment about only supporting values in {-1,0,1}.

Reply:
I think I've sorted this out now....